src/test/java/ro/developmentfactory/thedrones/service/DronePositionServiceImplTest.java

```diff
@@ -0,0 +1,267 @@
+package ro.developmentfactory.thedrones.service;
+
+import jakarta.persistence.EntityNotFoundException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
+import org.junit.jupiter.params.provider.EnumSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import ro.developmentfactory.thedrones.controller.dto.DroneStatusResponse;
+import ro.developmentfactory.thedrones.controller.dto.TurnDirection;
+import ro.developmentfactory.thedrones.repository.DroneRepository;
+import ro.developmentfactory.thedrones.repository.DroneStatusRepository;
+import ro.developmentfactory.thedrones.repository.entity.Direction;
+import ro.developmentfactory.thedrones.repository.entity.Drone;
+import ro.developmentfactory.thedrones.repository.entity.DroneStatus;
+
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.*;
+
+class DronePositionServiceImplTest {
+
+    @Mock
+    private DroneStatusRepository droneStatusRepository;
+
+    @Mock
+    private DroneRepository droneRepository;
+
+    @InjectMocks
+    private DronePositionServiceImpl dronePositionService;
+
+    private Drone drone;
+    private DroneStatus droneStatus;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+        droneStatus = new DroneStatus();
+        droneStatus.setCurrentPositionX(5);
+        droneStatus.setCurrentPositionY(5);
+        droneStatus.setFacingDirection(Direction.N);
+
+        drone = new Drone();
+        drone.setIdDrone(UUID.randomUUID());
+        drone.setDroneStatus(droneStatus);
+        drone.setCountMove(0);
+
+        droneStatus.setDrone(drone);
+    }
+
+    @Test
+    @DisplayName("Turning Direction: Valid right turn should update direction")
+    void turningDirection_ValidRightTurn_ShouldUpdateDirection() {
```

Comment:
you don't need this test because `turningDirection_ValidRotation_ShouldUpdateDirectionCorrectly` already covers it